tools/common/StringUtils.cs

```diff
@@ -48,6 +51,60 @@ public static string Quote (string f)
 			return s.ToString ();
 		}
 
+		public static string [] QuoteForProcess (IList<string> arguments)
+		{
+			if (arguments == null)
+				return Array.Empty<string> ();
+			return QuoteForProcess (arguments.ToArray ());
+		}
+
+		public static string [] QuoteForProcess (params string [] array)
+		{
+			if (array == null || array.Length == 0)
+				return array;
+
+			var rv = new string [array.Length];
+			for (var i = 0; i < array.Length; i++)
+				rv [i] = QuoteForProcess (array [i]);
+			return rv;
+		}
+
+		// Quote input according to how System.Diagnostics.Process needs it quoted.
+		public static string QuoteForProcess (string f)
+		{
+			if (String.IsNullOrEmpty (f))
+				return f ?? String.Empty;
+
+			if (f.IndexOfAny (mustQuoteCharactersProcess) == -1)
+				return f;
+
+			var s = new StringBuilder ();
+
+			s.Append ('"');
+			foreach (var c in f) {
+				if (c == '"') {
```

Comment:
This is quite short and simple (only 2 conditions), so I don't think a switch would make the code better. It would be at least two more lines in total, because of the break statements 😄